python_modules/dagster/dagster/_core/storage/batch_asset_record_loader.py

```diff
@@ -23,6 +23,11 @@ def add_asset_keys(self, asset_keys: Iterable[AssetKey]):
         unfetched_asset_keys = set(asset_keys).difference(self._asset_records.keys())
```

Comment:
`Set.difference(KeysView)` does not have a fast path implementation so it has to hash every key object even though the internal dict structure is based on hashes 

using a set comprehension against the dict directly is efficient though `unfetched_asset_keys = {asset_key for asset_key in asset_keys if asset_key not in self._asset_records}`

tested with:
```
import cProfile
from dagster import AssetKey
import random
import string


def random_key():
    # create a random name from string.asci_letters
    return [
        "".join(random.choices(string.ascii_letters, k=10)),
        "".join(random.choices(string.ascii_letters, k=10)),
        "".join(random.choices(string.ascii_letters, k=10)),
    ]


n = 1000

key_dict = {}
key_set = set()

for _ in range(n):
    key_dict[AssetKey(random_key())] = AssetKey(random_key())
    key_set.add(AssetKey(random_key()))

needle = AssetKey(random_key())
needles = (needle,)

with cProfile.Profile() as pr:
    set(needles).difference(key_dict.keys())
pr.print_stats(sort="ncalls")

with cProfile.Profile() as pr:
    set(needles).difference(key_set)
pr.print_stats(sort="ncalls")

with cProfile.Profile() as pr:
    set(needles).difference(set(key_dict.keys()))
pr.print_stats(sort="ncalls")

with cProfile.Profile() as pr:
    {n for n in needles if n not in key_dict.keys()}
pr.print_stats(sort="ncalls")
```

Reply:
oh man its even simpler, just don't call `.keys()` 

```
        unfetched_asset_keys = set(asset_keys).difference(self._asset_records)
```